sig-and-wg/sig-core/proposals/update-code-and-chart-in-one-pr.md

```diff
@@ -0,0 +1,84 @@
+# Update component and its chart in one pull request
+
+Created on 2019-23-04 by Adam Szecowka (@aszecowka) and Pawel Kosiec (@pkosiec). 
+
+## Status
+Proposed on 2019-23-04.
+
+## Motivation
+Currently, when a developer introduces changes to his component, he needs to do at least 2 PR:
+
+- Changes made to the component
+- Bump image version in the chart
+
+Such an approach has many drawbacks:
+
+- Slow development cycle, dealing with 2 PRs, asking for approvals twice
+- A developer can forget to bump image and version can be updated only on the release day. Updating version on the release day is 
+the worst case scenario because we postponed integration of our components. This is against Continuous Integration approach.
+- The code in the repository does not reflect what is really running.
+- Ugly repository history. Many commits only bump the component's image version.
+
+## Goal
+- Kyma developer can introduce his changes to Kyma in one Pull Request. Changes to the component and helm chart where we update component version is done in the same PR.
+- The code in the Kyma repository reflects code executed on the Kyma.
+When doing a release we rebuild all components. We should not be surprised on the release day that some tests are failing because some component was not updated for 3 weeks
+- Kyma repository has a nice commit history and commits that only bumps a component version are rare.
+- Run Kyma integration tests with the modified component within the same pull request status checks
+
+
+## Proposed solution
```

Comment:
Describe how the idea will work if there are to branches (and PRs) created for the same component at the same time.

Reply:
Added:
> In situation, that there are two PRs changing the same component, there will be a merge conflict for the second PR because
the same line in `values.yaml` was modified. Thanks to that, we ensure that `master` branch contains changes from both PRs.